### DISCUSSION TITLE
Class issues. fastapi registers self as a query parameter.

### DISCUSSION THREAD

**Discussion by lpmrfentazis** [fastapi/fastapi#10860] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, APIRouter, Depends, Form
from pydantic import BaseModel
from datetime import datetime

import uvicorn


class TokenInfoModel(BaseModel):
    token: str
    tokenType: str = "Barrer"
    expire: datetime

# Could be anything, in my code it's an ORM model
class UserDB:
    __tablename__ = 'users'

    id: int
    name: str
    username: str
    password: str
    active: bool
    roleID: int

    def checkPassword(self, password: str | bytes) -> bool:
        return True

class App:
    def __init__(self):
        # auth
        self.authRouter = APIRouter(prefix="/jwt", tags=["JWT"])
        self.authRouter.add_api_route("/login", self.login, methods=["POST"])

        self.server = FastAPI(swagger_ui_parameters={"syntaxHighlight.theme": "obsidian"})
        self.server.include_router(self.authRouter)

    async def validateUser(self, username: str = Form(), password: str = Form()):        
        ...

    # FIXME fastapi registers here self as a request parameter
    async def login(self, user: UserDB = Depends(validateUser)) -> TokenInfoModel:
        
        return TokenInfoModel("", datetime.utcnow())
    
    def start(self):
        uvicorn.run(self.server, host="0.0.0.0", port=8888)

if __name__ == "__main__":
    app = App()
    app.start()
```

### Description

I want to use a class based view, but not quite in the standard way. In my case, the custom class should fully serve the server controlling the device so that it can be used as an embedded module if needed.

I have previously discovered https://github.com/tiangolo/fastapi/issues/2220
But it seems the solution presented there is not suitable for me.

I am using APIRouter and add_api_route, and when I add a POST method using Depend I get that the self argument of my custom class is registered as a request parameter.

![image](https://github.com/tiangolo/fastapi/assets/43905097/2dac4498-53a8-4db6-a07e-6befd3ea3d7a)

### Operating System

Windows

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

3.11.3

**Comment by lpmrfentazis** [comment]
In general, the problem is solved if you make login and ValidateUser functions, for example inside init. But it doesn't seem to be the best solution.

```from fastapi import FastAPI, APIRouter, Depends, Form
from pydantic import BaseModel
from datetime import datetime

import uvicorn


class TokenInfoModel(BaseModel):
    token: str
    tokenType: str = "Barrer"
    expire: datetime

# Could be anything, in my code it's an ORM model
class UserDB:
    __tablename__ = 'users'

    id: int
    name: str
    username: str
    password: str
    active: bool
    roleID: int

    def checkPassword(self, password: str | bytes) -> bool:
        return True

class App:
    def __init__(self):
        # auth
        async def validateUser(self, username: str = Form(), password: str = Form()):        
            ...

        # FIXME fastapi registers here self as a request parameter
        async def login(self, user: UserDB = Depends(validateUser)) -> TokenInfoModel:
            
            return TokenInfoModel("", datetime.utcnow())
        
        self.authRouter = APIRouter(prefix="/jwt", tags=["JWT"])
        self.authRouter.add_api_route("/login", login, methods=["POST"])

        self.server: FastAPI = FastAPI(swagger_ui_parameters={"syntaxHighlight.theme": "obsidian"})
        self.server.include_router(self.authRouter)
    
    def start(self):
        uvicorn.run(self.server, host="0.0.0.0", port=8888)

if __name__ == "__main__":
    app = App()
    app.start()    
```

**Comment by YuriiMotov** [comment]
I think you can solve this using `partial` as well.

Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
